web/src/portico/landing-page.ts

```diff
@@ -94,14 +110,14 @@ const apps_events = function () {
 
         $(".info .platform").text(version_info.alt);
         $(".info .description").text(version_info.description);
-        $desktop_download_link.attr("href", version_info.download_link);
-        $download_from_google_play_store.attr("href", version_info.play_store_link);
-        $download_from_apple_app_store.attr("href", version_info.app_store_link);
-        $download_android_apk.attr("href", version_info.download_link);
-        $download_mac_arm64.attr("href", version_info.mac_arm64_link);
+        $desktop_download_link.attr("href", version_info.download_link!);
```

Comment:
@timabbott For example take a look at these lines. Here I'm using type assertion to tell typescript that these fields won't be `undefined`, but they actually will be.

Those fields will definitely be `undefined` for some iteration of a particular `OS` in info object. So, whether setting `href` attribute to `undefined` is expected behaviour I can't say. Because typescript does complain about it so it definitely is a bad practice.

Reply:
Thanks for flagging. We should avoid lying to TypeScript, and there’s no good reason to set `href` to `undefined` (actually it was coercing to the string `"undefined"`), so this should be guarded with `if (version_info.download_link !== undefined)`.

Reply:
@andersk Okay will do so.

Reply:
@andersk Take a look. 

One thing that's confusing is that there really isn't a need to set `href` conditionally, they can be set all at once, at initial render, only thing that needs to be changed is display logic. The links will always have the corresponding href attribute defined, we just need to toggle hide/unhide them according to url path/parameters.